### PR TITLE
Bug 1489306: Change to iframeless FxA form

### PR DESCRIPTION
**Pull request by stephaniehobson**
## Description
- Remove custom state JS and rely on fxa.js for states
  - Includes removing app store badges to use default download button
- Swap FxA iframe for form

## Issue / Bugzilla link
[Bug 1489306](https://bugzilla.mozilla.org/show_bug.cgi?id=1489306)
mozilla/bedrock#6149

## Testing
Different page states:
- Firefox signed in
- Firefox signed out
- Legacy Firefox (I tested this, it's probably okay for you not to)
- Firefox iOS
- Firefox Android
- Mobile, not Firefox
- Desktop, not Firefox

### CODE REVIEW

File: media/static-bundles.json [hunk 1]
```diff
@@ -1366,9 +1366,9 @@
     },
     {
       "files": [
-        "js/base/mozilla-fxa-iframe.js",
-        "js/base/uitour-lib.js",
-        "js/firefox/accounts.js"
+        "js/base/mozilla-fxa.js",
+        "js/base/mozilla-fxa-init.js",
+        "js/base/mozilla-fxa-form.js",
```
Comment (craigcook) [inline media/static-bundles.json]: This trailing comma is throwing a JSON error

**Comment by craigcook** [comment]
When signed in the status message is off-center because it's floated right. It's most apparent in a wide viewport, not so noticable when the box is narrower. I think just unfloating it will center it.

image

**Comment by craigcook** [comment]
I get the dreaded "Float Drop" at 1000px (the `$mq-desktop` breakpoint) until I hit about 1160px and there's room for them to float side by side.

image

**Comment by shane-tomlinson** [comment]
@stephaniehobson - is the request to /metrics-context being blocked with a CORS error? If so, @vladikoff is working on a fix in https://github.com/mozilla/fxa-content-server/issues/6535. 

Do you have an idea of when this PR will be shipped to prod? It'd be good if we could have our side shipped for when you ship this side.

**Comment by stephaniehobson** [comment]
Updated.

**Comment by stephaniehobson** [comment]
@shane-tomlinson Yes, it's the CORS request that's being blocked.

It will probably go to production tomorrow unless you'd like to ask for it to be delayed?

**Comment by vladikoff** [comment]
> Yes, it's the CORS request that's being blocked.

@stephaniehobson could you link to a stage or a prod deploy where we can see that?

**Comment by stephaniehobson** [comment]
@vladikoff Here's the new form on a demo sever: https://bedrock-demo-shobson.oregon-b.moz.works/en-US/firefox/accounts/

Note that it is submitting to your production servers (Jon had set up previous demos to go to your staging(?) servers. This one does not).

**Comment by shane-tomlinson** [comment]
> Note that it is submitting to your production servers (Jon had set up previous demos to go to your staging(?) servers. This one does not).

I think this is the reason the call to /flow-metrics is failing, prod has only enabled CORS for www.mozilla.org and www.allizom.org. Any way to get this onto www.allizom.org?

If not, our stage server enables CORS for /flow-metrics on:

- "https://www-demo1.allizom.org"
- "https://www-demo2.allizom.org"
- "https://www-demo3.allizom.org"
- "https://www-demo4.allizom.org"
- "https://www-demo5.allizom.org"
- "https://www-dev.allizom.org"
- "https://www.allizom.org"

**Comment by stephaniehobson** [comment]
I'll see if one of the numbered demo servers are free.

**Comment by stephaniehobson** [comment]
@shane-tomlinson Same error on demo4 from https://www-demo4.allizom.org/en-US/firefox/accounts/ the accounts dev server.

**Comment by stephaniehobson** [comment]
This started as an urgent emergency request on a Friday afternoon. Since the iframe is no longer throwing a 500 error it's no longer urgent and I recommend we hold merging it until we're happy with the behaviour on a dev server.

**Comment by stephaniehobson** [comment]
@craigcook The RTL fix is in but please do not merge once you approve.

**Comment by vladikoff** [comment]
@stephaniehobson demo4 should be now fixed 👍

**Comment by alexgibson** [comment]
Drive-by comment: I think there are functional tersts for this page that may need updating: https://github.com/mozilla/bedrock/blob/master/tests/functional/firefox/test_accounts.py

**Comment by shane-tomlinson** [comment]
Woohoo! When will this be shipped to prod?

**Comment by stephaniehobson** [comment]
Shipping now. Takes about an hour.

**Pull request merged**